src/components/pages/search.js

```diff
@@ -1,26 +1,49 @@
-import React, { useMemo } from 'react';
+import React, { useMemo, useState } from 'react';
 import Page from '../page.js';
 import PropTypes from 'prop-types';
 import moment from 'moment';
 import firstBy from 'thenby';
-
-import { Table, Tag } from 'antd';
+import { SearchOutlined } from '@ant-design/icons';
+import { Table, Tag, Select, Form } from 'antd';
 import { useTranslation } from 'react-i18next';
 import { searchResultShape } from '../../ui/default-prop-types.js';
 import { useLanguage } from '../language-context.js';
+import { useRequest } from '../request-context.js';
 import urls from '../../utils/urls.js';
 
 function Search({ initialState }) {
   const { t } = useTranslation('search');
   const { locale } = useLanguage();
   const { docs } = initialState;
+  const { query } = useRequest();
+
   const sortedDocs = useMemo(
-    () => docs.sort(firstBy(doc => doc.contributors)
-      .thenBy(doc => doc.updatedOn, 'desc')),
+    () => docs
+      .map(doc => ({
+        ...doc,
+        tagsSet: new Set(doc.tags)
+      }))
+      .sort(firstBy(doc => doc.tagMatchCount, 'desc')
+        .thenBy(doc => doc.updatedOn, 'desc')),
     [docs]
   );
 
-  const renderContributorsCount = value => (<div>{value?.length}</div>);
+  const [filteredDocs, setFilteredDocs] = useState([...sortedDocs]);
+
+  const [selectedTags, setSelectedTags] = useState([]);
+
+  const allTags = docs.reduce((acc, doc) => {
+    doc.tags.forEach(tag => acc.add(tag));
+    return acc;
+  }, new Set());
+
+  const handleTagsChanged = selectedValues => {
```

Comment:
Small thing: maybe `selectedTags` would be better here.

Reply:
That would conflict with the state variable "selectedTags"